src/libraries/Common/src/System/Security/Cryptography/MLKemPkcs8.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Formats.Asn1;
+using System.Security.Cryptography.Asn1;
+
+namespace System.Security.Cryptography
+{
+    internal static class MLKemPkcs8
+    {
+        internal static bool TryExportPkcs8PrivateKey(
+            MLKem kem,
+            bool hasSeed,
+            bool hasDecapsulationKey,
```

Comment:
Two bools is... fine, but every time I saw it I was thinking "this is really just one value".

```C#
internal enum SeededKeyState
{
    PublicOnly = 0,
    PrivateOnly = 1,
    Seed = 2,
}
```

(or flip 1 and 2).

Because `hasSeed:true` implies `hasDecapsulationKey: true`.

Yeah, the ldfld is cheap, and the registers are below the standard callconv count... but it's a thought I had every time, and decided to hold my tongue on.... until right before hitting approve, I guess :smile:

Reply:
Fair enough, I suppose :smile: